### PR TITLE
JSDoc-compatible code organization & API documentation

### PR DESCRIPTION
**Pull request by sebn**
- No documentation built in git yet (so we can still think about it)
- Can still be built locally with `yarn jsdoc`
- The `README.md` is not shorter/clearer
- The new `CODE_ORGANIZATION.md` is useful even without jsdoc
- All module comments are now jsdoc-compatible
- A few default export comments were moved
- A few comments were added

---

- [ ] PR is not too big
- [x] it improves UX & DX in some way
- [ ] it includes unit tests matching the implementation changes
- [ ] it includes scenarios matching a new behaviour or has been manually tested
- [x] it includes relevant documentation

### CODE REVIEW

File: core/local/index.js [hunk 1]
```diff
@@ -124,15 +132,17 @@ module.exports = class Local /*:: implements Side */ {
...
-  // This function updates utime and ctime according to the last
-  // modification date.
+  /**
+   * Return a function that will update last modification date
+   * and does a chmod +x if the file is executable
+   *
+   * Note: UNIX has 3 timestamps for a file/folder:
+   * - atime for last access
+   * - ctime for change (metadata or content)
+   * - utime for update (content only)
```
Comment (taratatach) [inline core/local/index.js]: is it `utime` or `mtime`?

File: test/unit/merge.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const pouchHelpers = require('../support/helpers/pouch')
 const dbBuilders = require('../support/builders/db')
 const Builders = require('../support/builders')
 
-/** Resolves with an object describing the side-effects of a Merge call.
+/* Resolves with an object describing the side-effects of a Merge call.
```
Comment (taratatach) [inline test/unit/merge.js]: Why remove the second star?

Comment (sebn) [reply test/unit/merge.js]: It is not importable/usable outside of this file.

Comment (sebn) [reply test/unit/merge.js]: Well, let's bring it back.

File: core/local/index.js (hunk 1 above)
Comment (sebn) [reply core/local/index.js]: Actually this comment is not accurate at all: `utimes` changes `atime`, not `ctime` :man_facepalming: 

But I don't think I changed anything here (except moving stuff around).
So let's fix it in another PR.

**Pull request merged**